helix-term/src/commands/typed.rs

```diff
@@ -1225,6 +1225,24 @@ fn lsp_restart(
     Ok(())
 }
 
+fn lsp_stop(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (_view, doc) = current!(cx.editor);
+    let config = doc
+        .language_config()
+        .context("LSP not defined for the current document")?;
+
+    cx.editor.language_servers.stop(config);
```

Comment:
This is missing a whole bunch of cleanup: all documents that use this language server will still refer to it and keep it alive. You'd need to scan over all documents and remove the language server. (There's somewhat similar bits of code in editor.rs)

Reply:
Done, hope the cleanup strategy is correct